src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,48 +3,162 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+
+    private static class Node<T> {
+        private T element;
+        private Node<T> first;
+        private Node<T> last;
+
+        public Node(Node<T> first, T element, Node<T> last) {
+            this.element = element;
+            this.first = first;
+            this.last = last;
+        }
+    }
+
+    private Node<T> head;
+    private Node<T> tail;
+    private int sizeList = 0;
+
     @Override
     public boolean add(T value) {
-        return false;
+        Node<T> lastNode = tail;
+        Node<T> newNode = new Node<>(lastNode, value, null);
+        tail = newNode;
+        if (lastNode == null) {
+            head = newNode;
+            sizeList++;
+        } else {
+            lastNode.last = newNode;
+            sizeList++;
+        }
+        return true;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index > sizeList || index < 0) {
+            throw new IndexOutOfBoundsException("Incorrect index");
+        }
+        if (index == sizeList) {
+            add(value);
+        } else {
+            linkBefore(value, nodeIndex(index));
+        }
     }
 
     @Override
     public boolean addAll(List<T> list) {
-        return false;
+        for (int i = 0; i < list.size(); i++) {
+            T value = list.get(i);
+            add(value);
+        }
+        return true;
     }
 
     @Override
     public T get(int index) {
-        return null;
+        isIndexExist(index);
+        return nodeIndex(index).element;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        isIndexExist(index);
+        Node<T> currentIndex = nodeIndex(index);
+        T oldVal = currentIndex.element;
+        currentIndex.element = value;
+        return oldVal;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        isIndexExist(index);
+        return position(nodeIndex(index));
     }
 
     @Override
     public boolean remove(T object) {
+        if (object == null) {
+            for (Node<T> newNode = head; newNode != null; newNode = newNode.last) {
```

Comment:
let's use a single loop independently if the object is null or not. You can compare it with null and non null values without risk of NPE if you write correct if condition